### ISSUE TITLE
Vue-3 editor contents disappearing before transitioning out

### ISSUE THREAD

**Issue by milesingrams** [ueberdosis/tiptap#2296]
### What’s the bug you are facing?

When using `@tiptap/vue-3` the `EditorContent` component will suddenly become empty if it is a child of an element transitioning out. I believe this is because the `onBeforeUnmount` hook in this code snippet:
https://github.com/ueberdosis/tiptap/blob/f24c7828c2d968b1f0c6652298a3160a0b47b558/packages/vue-3/src/EditorContent.ts#L56-L86

This hook is called when the leave transition is initiated thus emptying the contents. Vue 3 suggests calling this hook before the leave transition is expected behavior in this issue: https://github.com/vuejs/vue-next/issues/994

Thoughts/potential solutions:
1. Is it necessary to remove all child nodes of EditorContent? Will Vue not garbage collect them automatically?
2. If it is necessary can we perhaps provide a way to override the manual behavior and defer until the transition ends?

### How can we reproduce the bug on our side?

Here is the reproduction codepen:
https://codesandbox.io/s/tiptap-vue-3-transition-issue-zu9ol?file=/src/App.vue

### Can you provide a CodeSandbox?

Same as above:
https://codesandbox.io/s/tiptap-vue-3-transition-issue-zu9ol?file=/src/App.vue

### What did you expect to happen?

The editor should retain all nodes until the transition completes

### Anything to add? (optional)

TipTap is awesome and kinda blows my mind with how flexible it is. Thank you for such an awesome library! Will certainly sponsor if I release any project that uses it into to the wild wild web.

### Did you update your dependencies?

- [X] Yes, I’ve updated my dependencies to use the latest version of all packages.

### Are you sponsoring us?

- [ ] Yes, I’m a sponsor. 💖

**Comment by philippkuehn** [comment]
Hmm, as a workaround you can use `v-show="show"` instead of `v-if="show"`, since the editor is still only mounted once.

**Comment by milesingrams** [comment]
That might work for a different scenario but in our case we have a list (v-for) of cards that each contain a TipTap editor. The cards can be added and removed which means the parent element of the editor is unMounted and thus the editors unMounted hook is also called.

**Comment by milesingrams** [comment]
Here is a modified sandbox better representing the reason v-show wont work:
https://codesandbox.io/s/tiptap-vue-3-transition-issue-list-zu9ol?file=/src/App.vue

**Comment by milesingrams** [comment]
I managed to work around this in a funky way. I have a `ref` on the `EditorContent` element and before it unmounts I copy its HTML content and append it to the parent and then hide the original.

```javascript
onBeforeUnmount(() => {
	const staticEditorContent = document.createElement('div') // Create new div
	rootRef.value.append(staticEditorContent) // Attach to parent (i have a ref on it called rootRef)
	staticEditorContent.outerHTML = editorContentRef.value.$el.outerHTML // Set new element to contents of old one
	editorContentRef.value.$el.style.display = 'none' // Hide old element
	editor.value?.destroy() // Make sure this all happens before editor instance is destroyed
})
```

**Comment by pdesmarais** [comment]
👍 

Same issue here. It looks really weird when the editor is displayed in a modal. When the transition of the closing modal starts, the editor component disappears.

@milesingrams Thanks for your hacky workaround, at least I don't see the editor disappearing when I'm closing modals or changing pages. I hope we can remove this hack someday! 🤷‍♂️ 

Thanks for all the awesome work on TipTap! Just switched from QuillJS and we love it!

**Comment by cduque** [comment]
Sorry to bump this after so long but it happened to me too with a CSS transition. My workaround:

```
onBeforeUnmount(() => {
  const content = richText.value.innerHTML
  richText.value.innerHTML = content
  editor.destroy()
})
```

Being `richText` a `ref()` to the parent layer wrapping the editor component and `editor` my `Editor` instance.
Kudos to @milesingrams for the inspiration

HTH

**Comment by rabbitfufu** [comment]
Samesies.  The fix from @cduque works perfectly and is easy to implement.
I love tiptap and would never use anything else -- however it would be nice if this could be fixed at the level of the library.  (Without the fix, exit transitions are jarring.)

**Comment by nperez0111** [comment]
This is released with v2.9.0
